tests/ccl_test_cmblens.c

```diff
@@ -0,0 +1,91 @@
+#include "ccl.h"
+#include "../include/ccl_params.h"
+#include "ctest.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+#include <string.h>
+
+#define CLS_TOLERANCE 1E-3
+#define CLS_FRACTION 1E-3
+
+CTEST_DATA(cls) {
+  double Omega_c;
+  double Omega_b;
+  double h;
+  double A_s;
+  double n_s;
+  double sigma_8;
+};
+
+CTEST_SETUP(cls) {
+  data->Omega_c = 0.30;
+  data->Omega_b = 0.00;
+  data->h = 0.7;
+  data->A_s = 2.1e-9;
+  data->sigma_8=0.8;
+  data->n_s = 0.96;
+}
+
+static int linecount(FILE *f)
+{
+  //////
+  // Counts #lines from file
+  int i0=0;
+  char ch[1000];
+  while((fgets(ch,sizeof(ch),f))!=NULL) {
+    i0++;
+  }
+  return i0;
+}
+
+static void compare_cls(struct cls_data * data)
+{
+  int status=0;
+
+  ccl_configuration config = default_config;
+  config.transfer_function_method = ccl_bbks;
+  config.matter_power_spectrum_method = ccl_linear;
+  ccl_parameters params = ccl_parameters_create_flat_lcdm(data->Omega_c,data->Omega_b,data->h,
+							  data->A_s,data->n_s, &status);
+  params.Omega_k=0;
+  params.Omega_g=0;
+  params.Omega_n_rel=0;
+  params.Omega_l = 1.0 - params.Omega_m;
+  params.sigma_8=data->sigma_8;
+  ccl_cosmology * cosmo = ccl_cosmology_create(params, config);
+  ASSERT_NOT_NULL(cosmo);
+
+  //Fix spline parameters for the high-redshift needs of the CMB lensing power spectrum
+  ccl_splines->A_SPLINE_NA=10000;
+  ccl_splines->N_A=500;
+
+  char fname[256];
+  FILE *fi_cc;
+  CCL_ClTracer *tr_cl=ccl_cl_tracer_cmblens_new(cosmo,1090.,&status);
```

Comment:
I'm not sure whether we want z_source to be an input of the user in this way. Did you consider making it a parameter in the ini file or the cosmo struct?

Reply:
The correct thing to do here would be to query z_LSS from CLASS or something like that, but I couldn't figure out how to do that (or how to avoid calling CLASS if we just want the BBKS or E&H power spectra).

There's one advantage to having z_source as an external number: imagine that you want to compute the power spectrum of a source plane at any z, i.e. not just CMB lensing, but 21cm lensing or something like that...

The best solution would be probably to leave z_source as a user-defined variable, but allow for ccl_cosmology to compute z_LSS from cosmological parameters, which is the most common user case (i.e. users could compute z_LSS from ccl_cosmo and then pass it as an argument to compute C_ells).